tests/plugins/test_pulp_tag.py

```diff
@@ -0,0 +1,254 @@
+     .should_receive('copy')
+     .with_args(unicode, unicode))
+    (flexmock(dockpulp.Pulp)
+     .should_receive('updateRepo')
+     .with_args(unicode, dict))
+    (flexmock(dockpulp.Pulp)
+     .should_receive('crane')
+     .with_args(list, wait=True)
+     .and_return([2, 3, 4]))
+    (flexmock(dockpulp.Pulp)
+     .should_receive('')
+     .with_args(object, object)
+     .and_return([1, 2, 3]))
+    (flexmock(dockpulp.Pulp)
+     .should_receive('watch_tasks')
+     .with_args(list))
+
+    annotations = {
+        'worker-builds': {
+            'x86_64': {
+                'build': {
+                    'build-name': 'build-1-x64_64',
+                },
+                'metadata_fragment': 'configmap/build-1-x86_64-md',
+                'metadata_fragment_key': 'metadata.json',
+            },
+            'ppc64le': {
+                'build': {
+                    'build-name': 'build-1-ppc64le',
+                },
+                'metadata_fragment': 'configmap/build-1-ppc64le-md',
+                'metadata_fragment_key': 'metadata.json',
+            }
+        }
+    }
+
+    workflow.build_result = BuildResult(logs=["docker build log - \u2018 \u2017 \u2019 \n'"],
+                                        image_id="id1234", annotations=annotations)
+    build_info = {}
+    build_info['x86_64'] = BuildInfo()
+    build_info['ppc64le'] = BuildInfo()
+
+    for platform, v1_tag in v1_tags.items():
+        build_info[platform] = BuildInfo(v1_tag)
+
+    workflow.plugin_workspace = {
+        OrchestrateBuildPlugin.key: {
+            WORKSPACE_KEY_BUILD_INFO: build_info
+        }
+    }
+
+    mock_docker()
+    return tasker, workflow
+
+
+@pytest.mark.skipif(dockpulp is None,
+                    reason='dockpulp module not available')
+@pytest.mark.parametrize(("v1_tags", "should_raise"), [
+    ({'x86_64': None, 'ppc64le': None}, False),
+    ({'x86_64': None, 'ppc64le': 'ppc64le_v1_tag'}, False),
+    ({'x86_64': 'ppc64le_v1_tag', 'ppc64le': 'ppc64le_v1_tag'}, True),
+])
+def test_pulp_tag_basic(tmpdir, monkeypatch, v1_tags, should_raise, caplog):
+    tasker, workflow = prepare(check_repo_retval=0, v1_tags=v1_tags)
+    monkeypatch.setenv('SOURCE_SECRET_PATH', str(tmpdir))
+    with open(os.path.join(str(tmpdir), "pulp.cer"), "wt") as cer:
+        cer.write("pulp certificate\n")
+    with open(os.path.join(str(tmpdir), "pulp.key"), "wt") as key:
+        key.write("pulp key\n")
+
+    runner = PostBuildPluginsRunner(tasker, workflow, [{
+        'name': PLUGIN_PULP_TAG_KEY,
+        'args': {
+            'pulp_registry_name': 'test'
+        }}])
+
+    if should_raise:
+        with pytest.raises(PluginFailedException):
+            runner.run()
+        return
+
+    msg = None
+    for platform, v1_tag in v1_tags.items():
+        if v1_tag:
+            msg = "tagging v1-image-id {0} to platform {1}".format(v1_tag, platform)
+            break
+
+    runner.run()
+    assert PLUGIN_PULP_TAG_KEY is not None
```

Comment:
it's a straight translation of 
assert PulpPushPlugin.key is not None
from test_pulp.py

I confess I don't see the logic and will experiment with removing it.